src/api/security.py

```diff
@@ -0,0 +1,43 @@
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy.orm import Session
+
+from src.api.authorization import AuthHandler
+from src.api.schemas.users import UserCreate, UserLogin
+from src.database.models import UserModel
+from src.database.sql import get_db
+
+route = APIRouter()
+auth_handler = AuthHandler()
+
+
+@route.post('/register', status_code=201)
+async def register_new_user(user: UserCreate, db: Session = Depends(get_db)):
+    if db.query(UserModel).filter(UserModel.name == user.name).first():
+        raise HTTPException(status_code=400, detail=f'Username {user.name} is taken')
+    hashed_password = auth_handler.get_password_hash(user.password)
+    db.add(UserModel(name=user.name, email=user.email, password=hashed_password))
+    return {'msg': f'User {user.name} created'}
```

Comment:
It's bad idea to do many different operations in one place. You are checking a name and writing data to DB. Also don't see any password validations

Reply:
Reated a new module "service" and separated the operations in it. As for password validation, it is written in the user schema